scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -151,6 +153,17 @@ class FossId(
             v.takeUnless { k in secretKeys }.orEmpty()
         }
 
+    private suspend fun getProject(projectCode: String): Project? = service.getProject(user, apiKey, projectCode).run {
+        if (status == 0 && error == "Project does not exist") {
+            this@FossId.log.info { "Project does not exist." }
+            null
+        } else {
+            require(error == null && data != null) { "Could not get project. Additional information : $error" }
+            this@FossId.log.info { "Project '$projectCode' exists." }
+            data
+        }
```

Comment:
How about more explicitly separating the three cases "exists", "not exists", "don't know" like:

    private suspend fun getProject(projectCode: String): Project? =
        service.getProject(user, apiKey, projectCode).run {
            when {
                error == null && data != null -> {
                    FossId.log.info { "Project '$projectCode' exists." }
                    data
                }

                error == "Project does not exist" && status == 0 -> {
                    FossId.log.info { "Project '$projectCode' does not exist." }
                    null
                }

                else -> throw IOException("Could not get project '$projectCode'. Additional information: $error")
            }
        }